docs/ops/pooling/MaxPool_1.md

```diff
@@ -63,24 +63,192 @@
+  
+  * `auto_pad = explicit` and `rounding_type = ceil`  
+        `H_out = ceil(H + pads_begin[0] + pads_end[0] - kernel[0] / strides[0]) + 1`   
+        `W_out = ceil(W + pads_begin[1] + pads_end[1] - kernel[1] / strides[1]) + 1`    
+        `D_out = ceil(D + pads_begin[2] + pads_end[2] - kernel[2] / strides[2]) + 1`    
+  
+  * `auto_pad = valid` and `rounding_type = ceil`  
+        `H_out = ceil(H - kernel[0] / strides[0]) + 1`   
+        `W_out = ceil(W - kernel[1] / strides[1]) + 1`    
+        `D_out = ceil(D - kernel[2] / strides[2]) + 1`  
+  
+  * `auto_pad = same_upper/same_lower` and `rounding_type = ceil`  
+        `H_out = H`   
+        `W_out = W`    
+        `D_out = D`  
+
+If `H + pads_begin[i] + pads_end[i] - kernel[i]` is not divided by `strides[i]` evenly then the result is rounded with the respect to `rounding_type` attribute. 
+
+Example 1 shows how *MaxPool* operates with 4D input using 2D kernel and `auto_pad = explicit`
+
+```
+input = [[[[-1, 2, 3],
+           [4, 5, -6],
+           [-7, 8, 9]]]]
+strides = [1, 1]
+pads_begin = [1, 1]
+pads_end = [1, 1]
+kernel = [2, 2]
+rounding_type = "floor"
+auto_pad = "explicit"
+output = [[[[-1, 2, 3, 3],
+            [4, 5, 5, -6],
+            [4, 8, 9, 9],
+            [-7, 8, 9, 9]]]]
+```
+
+Example 2 shows how *MaxPool* operates with 3D input using 1D kernel and `auto_pad = valid`
+
+```
+input = [[[-1, 2, 3, 5, -7, 9, 1]]]
+strides = [1]
+kernel = [3]
+rounding_type = "floor"
+auto_pad = "valid"
+output = [[[3, 5, 5, 9, 9]]]
+```
+
+Example 3 shows how *MaxPool* operates with 4D input using 2D kernel and `auto_pad = same_lower`
+
+```
+input = [[[[-1, 2, 3],
+         [4, 5, -6],
+         [-7, 8, 9]]]]
+strides = [1, 1]
+kernel = [2, 2]
+rounding_type = "floor"
+auto_pad = "same_lower"
+output = [[[[-1, 2, 3],
+            [4, 5, 5]
+            [4, 8, 9]]]]
+```
+
+Example 4 shows how *MaxPool* operates with 4D input using 2D kernel and `auto_pad = same_upper`
+
+```
+input = [[[[-1, 2, 3],
+           [4, 5, -6],
+           [-7, 8, 9]],
+          [[2, -1, 5],
+           [6, -7, 1],
+           [8, 2, -3]]]]
+strides = [1, 1]
+kernel = [2, 2]
+rounding_type = "floor"
+auto_pad = "same_upper"
+output = [[[[5, 5, -6],
+            [8, 9, 9]
+            [8, 9, 9]],
+           [[6, 5, 5],
+            [8, 2, 1],
+            [8, 2, -3]]]]
+```
+
+Example 5 shows how *MaxPool* operates with 4D input using 2D kernel, `auto_pad = valid` and `rounding_type = ceil`
+
+```
+input = [[[[-1, 2, 3],
+           [4, 5, -6],
+           [-7, 8, 9]]]]
+strides = [2, 2]
+kernel = [2, 2]
+rounding_type = "ceil"
+auto_pad = "valid"
+output = [[[[5, 3],
+            [8, 9]]]]
+```
+
+**Examples**
 
 ```xml
 <layer ... type="MaxPool" ... >
-        <data auto_pad="same_upper" kernel="3,3" pads_begin="0,0" pads_end="1,1" strides="2,2"/>
-        <input> ... </input>
-        <output> ... </output>
+        <data auto_pad="same_upper" kernel="2,2" strides="2,2"/>
```

Comment:
There are much more required parameters according to the specification. Either make them optional or put them in this example

Reply:
@rkazants Told me to not place them there, because in same_upper mode they are not used

Reply:
Exactly, the other parameters are ignored in same_upper node so there is no need to have them in IR because it creates confusion.

Reply:
I'm with @lazarevevgeny on this, all required attributes should be present in examples.  If one is omitted but required this IR is illformed and cannot be read by IE and this can be even more confusing.  @rkazants or maybe you are saying that those parameters are in fact optional and IE reader will be able to read IR without them either way? In such case we could mark them as optional.

Reply:
@jdanieck, let me clarify what I meant. In case same_upper case there is no need to have `pads_begin`, `pads_end` and `strides` attributes and the IR will be still correct without them. If IE is currently unable to read such IR, this seems a bug there.
But of course we may leave these attributes in the IR, they will be ignored in this case as noted above and the IR will be still correct.

These attributes only makes sense for explicit padding mode. So I am leaning not to clutter IR with unneeded attributes.

Reply:
Ok. Got your point. I agree that semantically it makes not sense but I wasn't aware that we should handle such 'optionally required' attributes in IE:Reader. I can check how and if it works for MaxPool but it would be good to know what is the general rule here, because we'd also need to tests such IR's and I'm not sure if we are currently.  @ilyachur could you comment on this ?

Aside note is that we recently changed MO serializer to always put optional/defaulted attributes into IR (#3613). Not sure if you are aware of it.

Reply:
@lazarevevgeny @rkazants Do you know how the MO works for now?
And how is it aligned with approach that we need to add all attributes to IR?

Because in IE functional tests we have padding even if we have "same_upper" auto padd.
But if I remember right IE allows to read IRs without paddings.

Reply:
Guys, I just understood that these attributes (`pad_begin`, `pad_end` etc.) are mandatory in the specification. So the IR must have them in any case. Sorry for inconvinience. But it is incorrect because they should work only for explicit padding when you are explicitly saying how many pad values to insert on each corner. @ilyachur, `same_upper` automatically calculates a number of padding values on each corner. I presume that changing this attributes to optional is prohibited without raising operation version, isn't?

Reply:
I added pads_begin and pads_end to xml examples.